zulip_bots/zulip_bots/bots/googletranslate/googletranslate.py

```diff
@@ -0,0 +1,103 @@
+# To use this plugin, you need to set up the Google Cloud API key for this bot in
+# googletranslate.conf in this (zulip_bots/bots/googletranslate/) directory.
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+import requests
+
+class GoogleTranslateHandler(object):
+    '''
+    This bot will translate any messages sent to it using google translate.
+    Before using it, make sure you set up google api keys, and enable google
+    cloud translate from the google cloud console.
+    '''
+    def usage(self):
+        return '''
+            This plugin allows users translate messages
+            Users should @-mention the bot with the format
+            @-mention "<text_to_translate>" <target-language> <source-language(optional)>
+            '''
+
+    def initialize(self, bot_handler):
+        self.config_info = bot_handler.get_config_info('googletranslate')
+        self.supported_languages = get_supported_languages(self.config_info['key'])
+
+    def handle_message(self, message, bot_handler):
+        bot_response = get_translate_bot_response(message['content'],
+                                                  self.config_info,
+                                                  message['sender_full_name'],
+                                                  self.supported_languages)
+        bot_handler.send_reply(message, bot_response)
+
+api_url = 'https://translation.googleapis.com/language/translate/v2'
```

Comment:
I am a little vary of having all the bot code outside of the bot class. I would recommend creating a TranslateHelper class or similar for having this code instead. @roberthoenig what are your views on this?

Reply:
@aero31aero I'm not sure this is necessary, at least based on previous bots, though I would be interested to hear more thoughts. Ultimately the `Handler` class just needs the appropriate functions to be present, to be called from the library, so the implementation itself can go anywhere else that those functions can call?